lib/embed.dart

```diff
@@ -893,6 +914,24 @@ class Embed {
     dialog.showOk('Keyboard shortcuts', keyMapToHtml(keys.inverseBindings));
   }
 
+  WindowBase get _hostWindow {
+    if (window.parent != null) {
```

Comment:
Could this be a one-liner?

```
WindowBase get _hostWindow => window.parent ?? window;
```

If you prefer the current form for readability, keep it.